replit_river/client_session.py

```diff
@@ -83,10 +83,10 @@ async def send_upload(
         self,
         service_name: str,
         procedure_name: str,
-        init: Optional[InitType],
-        request: AsyncIterable[RequestType],
-        init_serializer: Optional[Callable[[InitType], Any]],
-        request_serializer: Callable[[RequestType], Any],
+        init: InitType,
+        request: Optional[AsyncIterable[RequestType]],
+        init_serializer: Callable[[InitType], Any],
+        request_serializer: Optional[Callable[[RequestType], Any]],
```

Comment:
Ideally the type is if `request` exists then `request_serializer` should not be optional.

Reply:
Maybe it should be `request: Optional[({request_type}, {request_serializer_type})]`